src/Workspaces/Core/Portable/PatternMatching/AllLowerCamelCaseMatcher.cs

```diff
@@ -134,6 +141,21 @@ private PatternMatchKind GetKind(CamelCaseResult result)
                 return bestResult;
             }
 
+            private static char ToLower(char v, TextInfo textInfo)
+            {
+                return IsAscii(v)
```

Comment:
TextInfo.ToLower is actually a bit slower here.  It checks IsLatin *and* IsAscii first, and it needs a virtual dispatch.  Considering that 99.999% of all strings we look at are ascii, we can just completely statically fastpath the heck out of this.  

Note: this helped for at leasta  couple percent improvement here.